superset/sql_lab.py

```diff
@@ -278,14 +279,17 @@ def _serialize_and_expand_data(
         ):
             data = (
                 pa.default_serialization_context()
-                .serialize(cdf.raw_df)
+                .serialize(result_table.pa_table)
                 .to_buffer()
                 .to_pybytes()
             )
+
         # expand when loading data from results backend
         all_columns, expanded_columns = (selected_columns, [])
     else:
-        data = cdf.data or []
+        df = result_table.to_pandas_df()
+        data = df_to_dict(df) or []
```

Comment:
shouldn't this be `df_to_dict(df) or {}`?

Reply:
One would think, but this is calling `to_dict(orient="records")` which returns list of dicts:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_dict.html

Reply:
Could be renamed `df_to_dicts` or `df_to_records` for clarity